src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2775,6 +2775,54 @@ LogicalResult ONNXDropoutOp::inferShapes() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// OneHotEncoder
+//===----------------------------------------------------------------------===//
+
+LogicalResult ONNXOneHotEncoderOp::inferShapes() {
+  ShapedType inputType = X().getType().dyn_cast<ShapedType>();
+  if (!X())
+    return emitError("no input found");
+
+  auto elementType = inputType.getElementType();
+  auto shape = inputType.getShape();
+  int64_t outDim = 0;
+
+  // If the input is a tensor of float, int32, or double,
+  // the data will be cast to integers and
+  // the cats_int64s category list will be used for the lookups.
+  if (elementType.isIntOrFloat()) {
+    if (!cats_int64s())
+      return emitError("input is a tensor of float, int32, or double, but no "
+                       "cats_int64s attribute");
+    outDim = ArrayAttrSize(cats_int64s());
+  } else {
+    if (!cats_strings())
+      return emitError("input is not a tensor of float, int32, or double, but "
+                       "no cats_strings attribute");
+    outDim = ArrayAttrSize(cats_strings());
+  }
+
+  // Encoded output data, having one more dimension than X
+  // total category count will determine the size of the extra dimension
+  int8_t flag = 0;
+  SmallVector<int64_t, 2> dims;
+  for (int i = 0; i != shape.size(); ++i) {
+    if (i == shape.size() - 1 && shape[i] == 1) {
```

Comment:
not sure whether this special case need to be "optimized" or not. Uniformly attach the one hot size to the last dimension may be better .